client/request.go

```diff
@@ -141,6 +141,13 @@ func (r *Request) WithJSONPayload(payload interface{}) *Request {
 	return r
 }
 
+func (r *Request) WithBody(contentType string, payload io.Reader) *Request {
+	r.header.Set(headers.ContentType, contentType)
```

Comment:
Perhaps its cleaner if we allow the user to set this with the WithHeader method instead?

and letting the WithBody just set the body of the request?

Reply:
I added it because the the contentType is set in the WithJsonPayload function going for similarity.

But I did try to use it by setting content-type header the regular way :/